core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java

```diff
@@ -80,6 +82,11 @@ private Settings nodeSettings(String dataPath) {
                 .build();
     }
 
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return pluginList(MockTransportService.TestPlugin.class);
```

Comment:
Does that matter here? The test would have already got this plugin randomized in anyways. At first I started trying to add in the ability to set it also on a per node basis, but there was way too much overhead to make it worthwhile (so many versions of create(async)node to pass it through. We could always move the test that needs it out to its own suite if it is a real concern?